tools/c7n_azure/c7n_azure/actions.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2015-2017 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Actions to take on Azure resources
+"""
+from c7n.actions import BaseAction
+from c7n import utils
+from c7n.filters import FilterValidationError
+from azure.mgmt.resource.resources.models import GenericResource
+
+
+class Tag(BaseAction):
+    """Add tags to Azure resources
+    """
+
+    schema = utils.type_schema(
+        'tag',
+        **{
+            'tag': {'type': 'string'},
+            'value': {'type': 'string'},
+            'tags': {'type': 'object'}
+        }
+    )
+
+    def validate(self):
+        if self.data.get('tags') and self.data.get('tag'):
+            raise FilterValidationError(
+                "Can't specify both tags and tag, choose one")
+        return self
+
+    def process(self, resources):
+        session = utils.local_session(self.manager.session_factory)
+        client = session.client('azure.mgmt.resource.ResourceManagementClient')
+
+        for resource in resources:
+            id = resource['id']
+
+            tags = self.data.get('tags') or resource.get('tags', {})
```

Comment:
So if `tags` is specified tags are replaced, if a single tag is specified, it is added - we are discussing if both scenarios should only be additive.

Reply:
we've generally separate out add/change vs remove as actions, i think its fine to combine the actions, but given the multi-modal usage it would be good i think to separate out the parameters (tags, remove).

Reply:
@aluong went back and looked at the AWS approach and is reworking this to be separate, he'll get that in shortly.  #2232